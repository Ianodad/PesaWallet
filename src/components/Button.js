import React from 'react';
import Text from "../components/Text"
import {StyleSheet, TouchableOpacity} from 'react-native';

import colors from '../config/colors';


const AppButton = ({
  title,
  onPress,
  style,
  color = 'primary',
  fontSize = 16,
  width,
}) => {
  return (
    <TouchableOpacity
      style={[styles.button, style, width, {backgroundColor: colors[color]}]}
      onPress={onPress}>
      <Text style={[styles.text, fontSize]}>{title}</Text>
    </TouchableOpacity>
  );
};

export default AppButton;

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.primary,
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 15,
    // width: "100%",
    // marginVertical: 10,
  },
  text: {
    color: colors.white,
    textTransform: 'uppercase',
    fontWeight: 'bold',
  },
});
