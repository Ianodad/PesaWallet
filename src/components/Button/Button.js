import React from 'react';
import {StyleSheet, TouchableOpacity, Image} from 'react-native';
import {Neomorph} from 'react-native-neomorph-shadows';

import {
  responsiveHeight,
  responsiveWidth,
} from 'react-native-responsive-dimensions';
import defaultStyles from '../../config/styles';
import Text from '../Text';

const AppButton = ({
  title,
  onPress,
  style,
  iconStyle,
  image,
  textStyle,
  color = 'primary',
  buttonType,
  disabled,
}) => {
  // {console.log(style)}
  return (
    <>
      {buttonType ? (
        <TouchableOpacity onPress={onPress} disabled={disabled}>
          <Neomorph
            darkShadowColor={'#00008B'}
            lightShadowColor={defaultStyles.colors.white}
            // inner
            swapShadows
            style={{
              ...style,
              ...styles.neomorphStyle,
              ...styles.button,
              backgroundColor: defaultStyles.colors[color],
            }}>
            {image && (
              <Image
                style={[styles.icon, iconStyle]}
                resizeMode="contain"
                source={image}
              />
            )}
            <Text style={[styles.text, textStyle]}>{title}</Text>
          </Neomorph>
        </TouchableOpacity>
      ) : (
        <TouchableOpacity
          style={[
            styles.button,
            style,
            {backgroundColor: defaultStyles.colors[color]},
          ]}
          onPress={onPress}
          disabled={disabled}>
          {image && (
            <Image
              style={[styles.icon, iconStyle]}
              resizeMode="contain"
              source={image}
            />
          )}
          <Text style={[styles.text, textStyle]}>{title}</Text>
        </TouchableOpacity>
      )}
    </>
  );
};

export default AppButton;

const styles = StyleSheet.create({
  button: {
    borderRadius: 40,
    justifyContent: 'center',
    // alignItems: 'center',
    width: responsiveWidth(90),
    height: responsiveHeight(8),
  },
  text: {
    color: defaultStyles.colors.white,
    textTransform: 'uppercase',
    fontWeight: 'bold',
  },
  neomorphStyle: {
    borderRadius: 45,
    shadowRadius: 11,
    marginVertical: 10,
    // padding: 15,
    // backgroundColor: colors.white,
    shadowOpacity: 0.9,
    width: responsiveWidth(90),
    height: responsiveWidth(90),
    shadowOffset: {width: 1, height: 0},
  },
});
