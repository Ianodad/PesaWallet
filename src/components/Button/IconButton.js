import React from 'react';
import {StyleSheet, View, Image, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';
// import colors from '../../config/colors';
import defaultStyles from '../../config/styles';
// import Text from '../Text';
// import Info from './Info';

const IconButton = ({
  image,
  style,
  imageStyle,
  info,
  iconStyle,
  icon,
  iconSize,
  color = 'primary',
  onPress,
}) => {
  // console.log(detail)
  return (
    <TouchableOpacity
      style={[
        styles.button,
        style,
        {backgroundColor: defaultStyles.colors[color]},
      ]}
      onPress={onPress}>
      {icon && (
        <Icon style={[styles.icon, iconStyle]} name={icon} size={iconSize} />
      )}

      {image && (
        <Image
          style={[styles.logo, imageStyle]}
          resizeMode="contain"
          source={image}
        />
      )}
      {info}
    </TouchableOpacity>
  );
};

export default IconButton;

const styles = StyleSheet.create({
  button: {
    flex: 1,
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  logo: {
    flex: 1,
  },
  subTitle: {
    position: 'absolute',
    zIndex: 2,
  },
  info: {
    flex: 1,
    position: 'absolute',
    zIndex: 2,
  },
});
