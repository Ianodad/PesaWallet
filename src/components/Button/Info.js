import React from 'react';
import {StyleSheet, View} from 'react-native';
import {color} from 'react-native-reanimated';
import defaultStyles from '../../config/styles';
import Text from '../Text';

const Info = ({style, subTitle, subStyle, detail, detailStyle}) => {
  return (
    <View style={[styles.container, style]}>
      {subTitle && <Text style={[styles.subTitle, subStyle]}>{subTitle}</Text>}
      {detail && (
        <Text style={[styles.detailStyle, detailStyle]}>{detail}</Text>
      )}
    </View>
  );
};

export default Info;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'absolute',
  },
  subTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: 'white',
  },
  detailStyle: {
    fontWeight: 'bold',
    color: 'white',
  },
});
