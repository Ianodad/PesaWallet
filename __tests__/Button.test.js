import React from 'react';
import {shallow} from 'enzyme';
import Button from '../src/components/Button/Button';

const defaultProps = {
  title: 'title',
  onPress: jest.fn(),
  style: {margin: '10px'},
  iconStyle: {margin: '10px'},
  image: jest.mock('../src/assets/bootsplash_logo.png'),
  textStyle: {color: 'black'},
  color: 'primary',
  buttonType: true,
};

describe('<Button/>', () => {
  const fn = jest.fn();
  const image = jest.mock('../src/assets/bootsplash_logo.png');
  describe('Rendering with snapshot', () => {
    // should render with any outcome
    it('should match to snapshot', () => {
      const component = shallow(<Button />);
      expect(component).toMatchSnapshot();
    });
    //
    it('should render with title snapshot', () => {
      const component = shallow(<Button title={'Submit'} />);
      expect(component).toMatchSnapshot();
    });

    it('should render with function props', () => {
      const component = shallow(<Button title={'Submit'} onPress={fn} />);
      expect(component).toMatchSnapshot();
    });

    it('should render with buttonType true', () => {
      const component = shallow(
        <Button title={'Submit'} onPress={fn} buttonType={true} />,
      );
      expect(component).toMatchSnapshot();
    });

    it('should render with buttonType begin false', () => {
      const component = shallow(
        <Button title={'Submit'} onPress={fn} buttonType={false} />,
      );
      expect(component).toMatchSnapshot();
    });

    it('should render with image', () => {
      const component = shallow(
        <Button title={'Submit'} onPress={fn} image={image} />,
      );
      expect(component).toMatchSnapshot();
    });

    it('should render with image and button true', () => {
      const component = shallow(
        <Button
          title={'Submit'}
          onPress={fn}
          buttonType={true}
          image={image}
        />,
      );
      expect(component).toMatchSnapshot();
    });

    it('should render with with passed props', () => {
      const component = shallow(<Button {...defaultProps} />);
      expect(component).toMatchSnapshot();
    });
  });
});
