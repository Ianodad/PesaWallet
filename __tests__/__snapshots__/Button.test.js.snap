// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Rendering with snapshot should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`<Button/> Rendering with snapshot should render with buttonType begin false 1`] = `ShallowWrapper {}`;

exports[`<Button/> Rendering with snapshot should render with buttonType true 1`] = `ShallowWrapper {}`;

exports[`<Button/> Rendering with snapshot should render with function props 1`] = `ShallowWrapper {}`;

exports[`<Button/> Rendering with snapshot should render with image 1`] = `ShallowWrapper {}`;

exports[`<Button/> Rendering with snapshot should render with image and button true 1`] = `ShallowWrapper {}`;

exports[`<Button/> Rendering with snapshot should render with title snapshot 1`] = `ShallowWrapper {}`;

exports[`<Button/> Rendering with snapshot should render with with passed props 1`] = `ShallowWrapper {}`;
